image/oci/layout/oci_dest.go

```diff
@@ -29,11 +36,14 @@ type ociImageDestination struct {
 	impl.PropertyMethodsInitialize
 	stubs.IgnoresOriginalOCIConfig
 	stubs.NoPutBlobPartialInitialize
-	stubs.NoSignaturesInitialize
 
-	ref           ociReference
-	index         imgspecv1.Index
-	sharedBlobDir string
+	ref            ociReference
+	index          imgspecv1.Index
+	sharedBlobDir  string
+	sys            *types.SystemContext
+	manifestDigest digest.Digest
+	// blobsToDelete is a set of digests which may be deleted
```

Comment:
At least
```suggestion
	// blobsToDelete is a set of digests which may be deleted _if_ we find no other references to them; it’s safe to optimistically include entries which may have other references
```
and perhaps rename this to `blobsToConsiderDeleting` or something like that?